cp23/hpi-fun2/granny/problem.tex

```diff
@@ -0,0 +1,37 @@
+\input{template.tex}
+
+\begin{document}
+
+\makeheader
+
+Madeline has baked a delicious strawberry cake for Granny and now wants to bring it to her as fast as possible.
+Both Granny and Madeline live in Celeste National Park, where there are a $n$ bus stations.
+These stations are numbered incrementally and connected trough $m$ bus lines, each denoting a bus driving from a start to an end station.
+Looking on the bus plan, Madeline tries to figure out the fastest way to get to Granny with the cake.
+Thanks to the wonderful service at the National Park, there is always a bus available at a station - meaning Madeline won't have to wait at stations.
+
+
+In order to save even more time, Madeline can instantly dash from one station to another - but only if the station's numbers are adjacent to another.
+For example, she can dash from station 5 to either station 4 or station 6.
+Since dashing exhausts Madeline very much, she can only dash once.
+Madeline currently lives at the first station and Granny lives at the very last station, but since she already visited her in the past, she knows that there is a way for her to get there.
+Considering the bus plan, how fast can she reach granny?
+
+\paragraph*{Input}
+
+The first line contains two integers $n$ and $m$ ($2 \leq n \leq 10^6, 1 \leq m \leq 5 * 10^3$): the number of bus stations and connections.
```

Comment:
```suggestion
The first line contains two integers $n$ and $m$ ($2 \leq n \leq 10^6, 1 \leq m \leq 5 \cdot 10^3$): the number of bus stations and connections.
```
You sure that your limits are correct? 10^6 is a bit tight on our judge and why is the edge bound so tight?